integration/regression_tests.sh

```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
+
+# Basic binary compatibility test script for um2nc.
+# See INTEGRATION_README.md for details on test usage, data and options.
+#
+# -----------------------------------------------------------------
+# Prepare options
+# -----------------------------------------------------------------
+
+function usage {
+ cat <<- EOF
+Basic binary compatibility test script for 'um2nc'.
+Compares 'um2nc' output against previous versions.
+
+Usage: regression_tests.sh [--keep] [-d DATA_CHOICE] [-v DATA_VERSION]
+
+Options:
+-k, --keep            Keep output netCDF data upon test completion. 
+                      If absent, output netCDF data will only be kept for failed test sessions.
+-d    DATA_CHOICE     Choice of test reference data. 
+                      Options: "full", "intermediate", "light".
+                      Default: "intermediate".
+-v    DATA_VERSION    Version of test reference data to use. 
+                      Options: "0".
+                      Default: latest release version.
+EOF
+}
+
+
+
+TEST_DATA_PARENT_DIR=/g/data/vk83/testing/um2nc/integration-tests
+
+# Default values, overwritten by command line arguments if present:
+TEST_DATA_CHOICE_DEFAULT=intermediate
+TEST_DATA_VERSION_DEFAULT=0
+CLEAN_OUTPUT=true
+
+while getopts ":-:d:hkv:" opt; do
+    case ${opt} in
+        -)
+            case ${OPTARG} in
+                help)
+                    usage
+                    exit 0
+                ;;
+                keep)
+                    CLEAN_OUTPUT=false
+                ;;
+                *)
+                    echo "Invalid option: \"--${OPTARG}\"." >&2
+                    usage
+                    exit 1
+                ;;
+            esac
+        ;;
+        d)
+            case ${OPTARG} in
+                full|intermediate|light)
+                    TEST_DATA_CHOICE=${OPTARG}
+                    ;;
+                *)
+                    echo "\"-${opt} ${OPTARG}\" is not a valid test data option. Choose between \"full\", \"intermediate\" and \"light\"." >&2
+                    usage
+                    exit 1
+                ;;
+            esac
+        ;;
+        h)
+            usage
+            exit 0
+        ;;
+        k)
+            CLEAN_OUTPUT=false
+        ;;
+        v)
+            TEST_DATA_VERSION=${OPTARG}
+        ;;
+        :)
+            echo "Option \"-${OPTARG}\" requires an argument." >&2
+            usage
+            exit 1
+        ;;
+        \?)
+            echo "Invalid option: \"-${OPTARG}\"." >&2
+            usage
+            exit 1
+        ;;
+    esac
+done
+
+# Check that no additional arguments were passed.
+if [[ -n "${@:$OPTIND:1}" ]]; then
+    echo "Invalid positional argument: \"${@:$OPTIND:1}\"." >&2
+    exit 1
+fi
+
+# Apply default data choice, version, and output directory if not set.
+TEST_DATA_CHOICE="${TEST_DATA_CHOICE:=$TEST_DATA_CHOICE_DEFAULT}"
+TEST_DATA_VERSION="${TEST_DATA_VERSION:=$TEST_DATA_VERSION_DEFAULT}"
```

Comment:
```suggestion
TEST_DATA_CHOICE="${TEST_DATA_CHOICE:-$TEST_DATA_CHOICE_DEFAULT}"
TEST_DATA_VERSION="${TEST_DATA_VERSION:-$TEST_DATA_VERSION_DEFAULT}"
```
Very minor, but in this case you might as well use parameter substitution and not assignment (since you're anyway assigning the same variable).

Reply:
Oh that's a good point, thanks!